cmd/daemon/player.go

```diff
@@ -195,6 +196,24 @@ func (p *AppPlayer) handlePlayerCommand(req dealer.RequestPayload) error {
 			return fmt.Errorf("failed shuffling context")
 		}
 
+		// Set queueID to the highest queue ID found in the queue.
+		// The UIDs are of the form q0, q1, q2, etc.
+		// Spotify apps don't seem to do this (they start again at 0 after
+		// transfer), which means that queue IDs get duplicated when tracks are
+		// added before and after the transfer and reordering will lead to weird
+		// effects. But we can do better :)
```

Comment:
I still feel like there should be a hidden field somewhere that shares this queue ID after a transfer, but considering the behavior with transfer between Spotify apps it doesn't seem so. It seems like such an obvious bug. Perhaps there is a field, but it got removed by go-librespot and never added again by any of the apps?

Reply:
Our protobuf definitions aren't always the most up-to-date (I usually update them when I figure out we are missing something) so there might be some missing fields we are not unmarshalling. You can have a look at the incoming transfer command and try to figure out if there's anything like it.

But, as you said, the fact that the Spotify official apps are bugged they might have not implemented it yet.

Reply:
There are indeed some fields that aren't included in the protobufs, but looking at the raw value they didn't seem anything like a queue ID. If I remember correctly, the fields stayed the same no matter how many songs I added to the queue.